src/beaconDecoder.py

```diff
@@ -0,0 +1,40 @@
+import socket
+import numpy as np
+import base64
+from receiveParser import ReceiveParser
+
+class beaconDecoder:
+
+    def __init__(self):
+        self.rxport = 4322
+        self.s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.s.connect(("127.0.0.1", self.rxport))
+        self.parser = ReceiveParser()
+    
+    def run(self):
+        while True:
```

Comment:
It would be better to refactor run() so that it receives one beacon packet and returns the dict. Then the caller (main in this case) will do the printing of the dict. Makes it more portable, since the website can't capture print statements but it can take the return value of dict and do what it wants from there. 
It would be more pythonic to implement something with yields, but we won't worry about that for now. 
Printing that a beacon is received isn't necessary either, we know one was received because a dict was returned. Invalid packet number could be an exception, or just silently ignore it.

Reply:
Makes sense.